### PR TITLE
Add Push Notification support (CR2483)

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

**Comment by Danielius1922** [comment]
@joochlee I had a hard time rebasing and squashing your original branch, so instead I've created a new one and copied the changed files. I've run diff on the branches and they are identical, so it should be OK to merge this one.

**Comment by joochlee** [comment]
> @joochlee I had a hard time rebasing and squashing your original branch, so instead I've created a new one and copied the changed files. I've run diff on the branches and they are identical, so it should be OK to merge this one.

It looks good, Thank you Daniel ! 👍

**Review by jkralik — CHANGES_REQUESTED** [review]
pls replace `rsc` by `resource` for API

File: apps/docs/push.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+		```
+
+### Target Server
+- `oc_set_on_push_arrived()`
+	```c
+	/**
+	 * @brief object used to store Resource pushed to
+	 * "oic.r.pshreceiver:receivers[i].receiveruri"
+	 */
+	typedef struct oc_pushd_rsc_rep
```
[the remark this answers is not in the document]
Comment (joochlee) [reply apps/docs/push.md:168]: do you want to change `struct oc_pushd_rsc_rep` into `struct oc_pushd_resource_rep` only, 
or `oc_pushd_rsc_rep_t` into `oc_pushd_resource_rep_t` too ?

[the remark this answers is not in the document]
Comment (joochlee) [reply apps/docs/push.md]: fixed:
- `oc_pushd_rsc_rep` => `oc_pushd_resource_rep`
- `oc_pushd_rsc_rep_t` => `oc_pushd_resource_rep_t`

[the remark this answers is not in the document]
Comment (jkralik) [reply apps/docs/push.md:168]: I mean to get rid of `rsc` abbreviation from types and functions in whole PR.

[the remark this answers is not in the document]
Comment (joochlee) [reply apps/docs/push.md:168]: > I mean to get rid of `rsc` abbreviation from types and functions in whole PR.

do you mean that just removing `rsc` from type name ? 
or replacing `rsc` with `resource` ?

I replaced `rsc` with `resource` in whole PR like as follows:
-  `oc_pushd_rsc_rep` => `oc_pushd_resource_rep`
-  `oc_pushd_rsc_rep_t` => `oc_pushd_resource_rep_t`

[the remark this answers is not in the document]
Comment (jkralik) [reply apps/docs/push.md]: Please renames macros and function:
```
PUSHCONF_RSC_*
oc_print_pushd_rsc
```

**Comment by Danielius1922** [comment]
> Thank you Daniel! +1+1
> 
> I have been busy for other works in last week...

I thought that was the case, but Jozef said that we should merge this today ( well, he actually said that on Friday :D ), so I renamed the public stuff to desired names (and did some other minor stylistic changes). The tests passed, so once Jozef approves I'll merge.
I've also created #319 , so we can fix the issues reported by SonarCloud and clang-tidy and add unit tests.

**Comment by joochlee** [comment]
> > Thank you Daniel! +1+1
> > I have been busy for other works in last week...
> 
> I thought that was the case, but Jozef said that we should merge this today ( well, he actually said that on Friday :D ), so I renamed the public stuff to desired names (and did some other minor stylistic changes). The tests passed, so once Jozef approves I'll merge. I've also created #319 , so we can fix the issues reported by SonarCloud and clang-tidy and add unit tests.

got it! 😎
Thanks

**Pull request merged**